File: java/challenges/SlotMachine.java

import java.util.*;

public class SlotMachine
{
  public static void main(String[] args)
  {
    Scanner keyboard = new Scanner(System.in);
    int money = enterAmountOfMoney(keyboard);
    String firstWord = getChoice();
    String secondWord = getChoice();
    String thirdWord = getChoice();
    checkWinnings(money, firstWord, secondWord, thirdWord);
    playAgain(keyboard);
    keyboard.close();
  }

  public static int enterAmountOfMoney(Scanner keyboard)
  {
    int money;
    System.out.println("\n\nPlease enter the amount of money you wish to bet");
    money = keyboard.nextInt();
    if (money < 0) {
      System.out.println("Please enter a while number greater than zero");
      money = enterAmountOfMoney(keyboard);
    }
    return money;
  }

  public static String getChoice()
  {
  int number = (int) (Math.random()*6);
  String choice = new String [] {"Cherries", "Oranges", "Plums", "Bells", "Melons", "Bars"}[number];
  System.out.println(choice);
  return choice;
  }

  public static void checkWinnings(int money, String firstWord, String secondWord, String thirdWord)
  {
    if (firstWord == secondWord && firstWord == thirdWord)
    {
      money *= 3;
      System.out.println("You won $" + money + "!");
    } else if (firstWord == secondWord || firstWord == thirdWord || secondWord == thirdWord)
    {
      money *= 2;
      System.out.println("You won $" + money + "!");
    } else
    {
      System.out.println("You loose!");
    }
  }

  public static void playAgain(Scanner keyboard)
  {
    System.out.println("Do you want to play again? Please enter 'Y' or 'y' to do so.");
    String answer;
    answer = keyboard.next();

    if (answer.charAt(0) == 'Y' || answer.charAt(0) == 'y')
    {
      System.out.println("Have fun!\n\n");
      main(new String[0]);
    } else
    {
      System.out.println("Bye Bye!");
    }
  }
}

File: python/challenges/slot_machine.py
import random

def main():
  money = enter_amount_of_money()
  first_word = get_choice()
  second_word = get_choice()
  third_word = get_choice()
  check_winnings(money, first_word, second_word, third_word)


def enter_amount_of_money():
  money = input("Please enter the amount of money you wish to bet\n")
  if money == '' or float(money) < 0:
    print("Please enter an amount greater than or equal to zero.\n")
    money = enter_amount_of_money()
  money = int(float(money))
  return money

def get_choice():
  word = random.sample(["Cherries", "Oranges", "Plums", "Bells", "Melons", "Bars"], 1)
  print(word)
  return word

def check_winnings(money, first_word, second_word, third_word):
  if first_word == second_word and first_word == third_word:
    money = money * 3
    print(f"You won ${money}!")
  elif first_word == second_word or first_word == third_word or second_word == third_word:
     money = money * 2
     print(f"You won ${money}!")
  else:
    print("You loose!")
  play_again()

def play_again():
  answer = input("Do you want to play again? Please enter 'Y' or 'y' to do so.\n")
  if answer == 'Y' or answer == 'y':
    print("Have Fun!\n\n\n")
    main()
  else:
    print("Bye Bye!")

main()